### PR TITLE
[core] Properties refactoring: Move static constants of ValueParser to class ValueParserConstants

### PR DESCRIPTION
**Pull request by oowekyala**
Before submitting a PR, please check that:
 - [x] The PR is submitted against `master`. The PMD team will merge back to support branches as needed.
 - [x] `mvn test` passes.
 - [x] `mvn checkstyle:check` passes. [Check this for more info](https://github.com/pmd/pmd/blob/master/CONTRIBUTING.md#code-style)

**PR Description:** Checks item 2 of #504

### CODE REVIEW

File: pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/properties/ValueParsers.java [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+import org.apache.commons.lang3.StringUtils;
+
+import net.sourceforge.pmd.util.ClassUtil;
+
+/**
+ * @author Clément Fournier
+ * @since 6.0.0
+ */
+//TODO: make package-private when we move everything to n.s.pmd.properties
+public final class ValueParsers {
```
Comment (jsotuyod) [inline pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/properties/ValueParsers.java]: I like the idea of moving these out, but I believe we need a different name.

In general, it's a a code smell to have classes differing on a single char, specially plurals

Comment (oowekyala) [reply pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/properties/ValueParsers.java]: Maybe `ValueParserConstants` then ?

Comment (jsotuyod) [reply pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/properties/ValueParsers.java]: That's a good name!

**Comment by jsotuyod** [comment]
Very late, but finally merged! Thanks!

**Pull request merged**